pkg/build/strategies/dockerfile/dockerfile.go

```diff
@@ -422,20 +418,29 @@ func getDestination(config *api.Config) string {
 	return destination
 }
 
-// getImageScriptsDir returns the directory containing the builder image scripts and a bool
-// indicating that the directory is expected to contain all s2i scripts
-func getImageScriptsDir(config *api.Config) (string, bool) {
+// getImageScriptsDir returns the default directory which should contain the builder image scripts
+// as well as a map of booleans identifying  individual scripts provided in the repository as overrides
+func getImageScriptsDir(config *api.Config, builder *Dockerfile) (string, map[string]bool) {
+
+	// 1st priority is the command line parameter (pointing to an image, overrides it all)
 	if strings.HasPrefix(config.ScriptsURL, "image://") {
-		return strings.TrimPrefix(config.ScriptsURL, "image://"), true
+		return strings.TrimPrefix(config.ScriptsURL, "image://"), make(map[string]bool)
 	}
+
+	// 2nd priority (the source code repository), collect the locations
+	providedScripts := scanScripts(filepath.Join(config.WorkingDir, builder.uploadScriptsDir))
+
+	// 3rd priority (the builder image), collect the locations
 	scriptsURL, _ := util.AdjustConfigWithImageLabels(config)
 	if strings.HasPrefix(scriptsURL, "image://") {
-		return strings.TrimPrefix(scriptsURL, "image://"), true
+		return strings.TrimPrefix(scriptsURL, "image://"), providedScripts
 	}
 	if strings.HasPrefix(config.ImageScriptsURL, "image://") {
-		return strings.TrimPrefix(config.ImageScriptsURL, "image://"), false
+		return strings.TrimPrefix(config.ImageScriptsURL, "image://"), providedScripts
```

Comment:
Shouldn't this return `make(map[string]bool)` instead according to it's previous use?

Reply:
I don't think so. This function has been expanded to merge the precedence order (which was the issue with the previous version). 

In this case, when scripts have been found in the local repo, but then in the 3rd and 4th priority cases where an annotation or fallback exists, we should merge/override with the scripts from the local repo.

The only correct spot to return an empty found map is when we have a command line parameter pointing to image:/// location (because we won't know and don't inspect the image to see *which* scripts may be there, so it's unilateral).

It'd be worthwhile in a future enhancement to understand more deeply how we should handle or inspect when file:/// or http:// are indicated for these ordered search locations. I couldn't see where that it's inspected anywhere and didn't try it.